src/InferenceFramework.cpp

```diff
@@ -10,17 +10,17 @@
 namespace EERAModel {
 namespace Inference {
 
-InferenceFramework::InferenceFramework(Model::ModelInterface::Sptr model,
-    const ModelInputParameters& modelInputParameters,
-    const InputObservations& observations,
+InferenceFramework::InferenceFramework(Model::ModelInterface::Sptr& model,
```

Comment:
I don't quite understand what's going on here. We've gone from passing a shared pointer to passing a __reference__ to a shared pointer. I think that sort of defeats the purpose of a shared pointer, which is to allow multiple units of code to share ownership of an item. With this change, there is really only a single pointer, maintained by the `main` function, and other code units are referring to it instead of maintaining their own pointer.

What was clang-tidy's argument for doing this?

Reply:
The warning from clang-tidy was: `warning: pass by value and use std::move [modernize-pass-by-value]`, changing to reference removed the warning without changing the regression test results (granted that doesn't guarantee correctness). I think given that the same warning was given for basically every other argument in the class definition, it may be fine to ignore this warning from clang-tidy and revert to the old style if shared ownership is the aim.